### PR TITLE
[DO-NOT-MERGE] more C++ bindings

### PR DESCRIPTION
**Pull request by skimo-openhub**
This extends the C++ bindings with subclasses for isl_schedule_node.

### CODE REVIEW

File: interface/generator.cc [hunk 1]
```diff
@@ -70,6 +70,20 @@ FunctionDecl *generator::find_by_name(const string &name, bool required)
 	return NULL;
 }
 
+/* Add a class derived from "decl" to the set of classes,
+ * keeping track of the _to_str, _copy and _free functions, if any, separately.
+ */
```
Comment (tobig) [inline interface/generator.cc]: Nice!

**Comment by skimo-openhub** [comment]
On Wed, Dec 06, 2017 at 06:48:41AM -0800, Tobias Grosser wrote:
> tobig commented on this pull request.
> 
> Hi Sven,
> 
> thanks for pushing the bindings forward. The general direction looks great and most patches are very solid. I added a couple of minor changes.
> 
> Here some high-level comments:
> 
> - Would it make sense to review these changes in public at https://github.com/PollyLabs/isl? This 
>   would allow others (Michael, Alex Isonard) to participate in the initial discussions, will make 
>   sure that all design decisions are publicly documented, and also gives us more visibility which helps 
>   prove the point that there is an actual community around polyhedral compilation.
> 
> - I feel these reviews are easier if we review feature branches rebased on the latest public isl. Rather 
>    than a "more C++", could we just have a review per feature? E.g. the operator bool() patch seems 
>    mostly independent. We could discuss it and push it separately. This would make the discussions 
>    more focused!

I'll do that when it's ready to go public, but I'm already
spending too much time on these bindings now.

skimo

**Comment by skimo-openhub** [comment]
On Wed, Dec 06, 2017 at 06:49:48AM -0800, Tobias Grosser wrote:
> tobig commented on this pull request.
> 
> You might also add a comment why we made the operator bool() explicit.

Done.

skimo

**Comment by skimo-openhub** [comment]
Superseded by #67.

**Pull request closed without merging**